### DISCUSSION TITLE
WP Rig v3 - Block Theme FSE Features

### DISCUSSION THREAD

**Discussion by Spleeding1** [wprig/wprig#776] [Ideas]
As WordPress is starting to standardize FSE and block themes, we begin to think about the next version of WP Rig.
[Block Theme](https://developer.wordpress.org/block-editor/how-to-guides/themes/block-theme-overview/)

What are some features you would like to see in development of the next WP Rig?
(please add each feature as its own comment so they can get voted on individually)

**Comment by Spleeding1** [comment]
I'll start off by saying PostCSS support

**Reply by dthenley** [reply]
We do already have PostCSS in the theme. The theme actually currently does a great job on implementing a good portion of future css which one of my favorite is nesting but depending on your editor it's not easy to implement right out the box. One thing we can do is figure out how to showcase the ability to nest css and other hidden gems from the theme.

**Reply by Spleeding1** [reply]
@dthenley Yes we do. If you haven't looked at the [block theme](https://developer.wordpress.org/block-editor/how-to-guides/themes/block-theme-overview/) page check it out. WP Rig will compile to a block theme, so it won't be compatible with WP Rig v2.

**Reply by dthenley** [reply]
Ah ok, so you are wanting to make sure it carries over to v3. Well one thing I love about wp rig and i think that makes it stand out more than others is that it is coming as a OOP php forward theme. I want to make sure we keep the functions broken out into pieces like we currently have. I understand that the theme.json will make a lot of it invalid but I'm sure there is tons of use cases where the theme.json isnt completely taking over the functions.php file. And to have the option to keep each component or feature seperated like it is currently is the best thing to me.

**Reply by Spleeding1** [reply]
Yes. I agree about isolation. Although, have you looked at a block theme functions.php? it's almost non-existent! Not sure how a lot of things work yet.

**Reply by LukaszJaro** [reply]
yup, you can see this starter theme structure also heavily relies on modules and author mentioned complications https://github.com/generaxion/axio-starter/issues/55

**Comment by dthenley** [comment]
A couple things i would like to have would be:

- Theme.json already pre built and integrated into the css of the site.
- Extending Block styles on a couple chosen blocks - https://developer.wordpress.org/block-editor/how-to-guides/javascript/extending-the-block-editor/
- Add Block filters options to be implemented to jumpstart that more complex editting feature - https://developer.wordpress.org/block-editor/reference-guides/filters/block-filters/#block-styles
- Have the theme come woocommerce ready as well

**Comment by LukaszJaro** [comment]
I don't yet know what I would want in a block theme but has there been discussion about removing wp-rigs progressive loading of CSS?

Full-site editing will take care of this in a standard way:

> The way CSS is loaded was improved significantly in Gutenberg version 9.6. Gutenberg now only loads the styles of the blocks that are used on a page.
> If you are interested in learning about the continued work to improve the performance, see https://github.com/WordPress/gutenberg/pull/28358
-https://fullsiteediting.com/lessons/global-styles/#h-how-does-global-styles-work

**Reply by Spleeding1** [reply]
It definitely will be discussed, but we just started this to get an idea of where it will go.

**Reply by dthenley** [reply]
Will it be applying only progressive loading the blocks, or does it do it for pages as well. I like the fact that we're able to preload css according to our own parameters and not just only to blocks

**Reply by Spleeding1** [reply]
Essentially everything is a block. WordPress FSE Theme by default no longer has CSS files. It generates on page load according to theme.json. CSS files can be used, but it I think it is discouraged. Not sure where media queries fit in yet.
